### PR TITLE
add caching for mapdata

### LINKED ISSUE #361: enable caching for /data
### Subject of the issue
auto generate  keralaresruce.in/data

### PR DESCRIPTION
**Pull request by ashwinrajeev**
### Issue Reference
This PR addresses the Issue : Fixes #361 

### Summarize
caches /data in redis

#### Notes
* added new env variables REDIS_URL, CACHE_TIMEOUT
* CACHE_TIMEOUT is in seconds

### CODE REVIEW

**Comment by tachyons** [comment]
Oops, I am 3 minute late!  https://github.com/IEEEKeralaSection/rescuekerala/pull/398. I'll close mine

File: mainapp/views.py [hunk 1]
```diff
@@ -168,11 +170,16 @@ class Maintenance(TemplateView):
...
+    data = cache.get("mapdata:" + district)
+    if data:
+        return JsonResponse(list(data) , safe=False)
+    if district != "all":
+        data = Request.objects.exclude(latlng__exact="").filter(district=district).values()
+        cache.set("mapdata:" + district, data, settings.CACHE_TIMEOUT)
     else:
         data = Request.objects.exclude(latlng__exact="").values()
-
+        cache.set("mapdata:" + district, data, settings.CACHE_TIMEOUT)
```
Comment (tachyons) [inline mainapp/views.py]: You can take this out side of `if else` block since you have same code for both cases

Comment (ashwinrajeev) [reply mainapp/views.py]: ok

**Comment by tachyons** [comment]
Heroku uses `REDIS_URL` `env` for Redis addon. So it is just matter of plug and play, No configuration required in heroku

**Comment by drtechie** [comment]
image
Tested
/data endpoint is working fine.

**Comment by cibingeorge** [comment]
We deploy this? @vigneshhari @biswaz

**Pull request merged**